### ISSUE TITLE
valid code getting Invalid argument name; Argument names must not be quoted

### ISSUE THREAD

**Issue by sdbruder** [terraform-linters/tflint#372]
the following minimal example fails on flint with `Error: Failed to load configurations: test.tf:4,5-6: Invalid argument name; Argument names must not be quoted.`

```
resource` "datadog_monitor" "example" {
  # For now silenced to avoid notifications.
  silenced {
    "*" = "0"
  }
}
```

**Comment by wata727** [comment]
Hi @sdbruder 

I confirmed the behavior, but it is the same with Terraform's behavior.

```
$ tflint
Error: Failed to load configurations: test.tf:4,5-6: Invalid argument name; Argument names must not be quoted.
$ terraform validate

Error: Invalid argument name

  on test.tf line 4, in resource "datadog_monitor" "example":
   4:     "*" = "0"

Argument names must not be quoted.
```

Is this code really valid? TFLint is compatible with Terraform 0.12.x, so if you assume to run against Terraform 0.11.x and older, you may get unexpected results.

**Comment by jeremygaither** [comment]
I can reproduce this with the following block, but `terraform validate` passes:

```hcl
resource "null_resource" "validate_name_length" {
  count = "${length(local.name_prefix) <= 32 ? 0 : 1}"

  "ERROR: name and env must be 32 chars or less" = true
}
```

**Comment by wata727** [comment]
TFLint is compatible with Terraform v0.12.x, so it is still invalid configuration even if `terraform validate` passes in Terraform v0.11.x ...

**Comment by jeremygaither** [comment]
Is TFLint no longer compatible with valid Terraform 0.11 code?

**Comment by jeremygaither** [comment]
Merging #390 makes tflint impossible to use for many of our 0.11 projects any longer.

**Comment by wata727** [comment]
Yes, if it is invalid at 0.12, it will not work. This is the same behavior as in TFLint v0.8.

**Comment by wata727** [comment]
Close this due to no activity.
